core/src/main/java/com/nisovin/magicspells/util/handlers/RecipeHandler.java

```diff
@@ -0,0 +1,208 @@
+                if (ingredient == null) return;
+
+                float experience = (float) config.getDouble("experience", 0);
+                int cookingTime = config.getInt("cooking-time", 0);
+
+                if (type.equals("furnace")) {
+                    FurnaceRecipe furnaceRecipe = new FurnaceRecipe(namespace, result, ingredient, experience, cookingTime);
+                    furnaceRecipe.setGroup(group);
+                    Bukkit.addRecipe(furnaceRecipe);
+                }
+                else {
+                    boolean added = MagicSpells.getVolatileCodeHandler().addCookingRecipe(type, namespace,group, result, ingredient, experience, cookingTime);
+                    if (!added) {
+                        MagicSpells.error("Recipe type '"+ type + "' on recipe '" + config.getName() + "' is unsupported on this version of spigot.");
+                        return;
+                    }
+                }
+                break;
+
+            case "stonecutting":
+                ingredient = getMaterial(config.getString("ingredient"), "Recipe '" + config.getName() + "' has an invalid 'ingredient' defined.");
+                if (ingredient == null) return;
+                boolean added = MagicSpells.getVolatileCodeHandler().addStonecutterRecipe(namespace, group, result, ingredient);
+                if (!added) {
+                    MagicSpells.error("Recipe type '"+ type + "' on recipe '" + config.getName() + "' is unsupported on this version of spigot.");
+                    return;
+                }
+                break;
+
+            default:
+                MagicSpells.error("Recipe '" + config.getName() + "' has an invalid 'type' defined.");
+                return;
+        }
+        namespaces.add(namespace);
+    }
+
+    private static ItemStack getItem(String itemString, String errorMsg) {
+        if (itemString == null || itemString.isEmpty()) {
+            MagicSpells.error(errorMsg);
+            return null;
+        }
+        ItemStack item = Util.getItemStackFromString(itemString);
+        if (item == null) {
+            MagicSpells.error(errorMsg);
+            return null;
+        }
+        return item;
+    }
+
+    private static Material getMaterial(String materialName, String errorMsg) {
+        if (materialName == null || materialName.isEmpty()) {
+            MagicSpells.error(errorMsg);
+            return null;
+        }
+        Material material = null;
+        try {
+            material = Material.getMaterial(materialName.toUpperCase());
+        }
+        catch (IllegalArgumentException ignored) {}
+        if (material == null) {
+            MagicSpells.error(errorMsg);
+            return null;
+        }
+        return material;
+    }
+
+    public static int count() {
+        return namespaces.size();
+    }
+
+    public static void resetAll() {
+        Bukkit.resetRecipes();
```

Comment:
WOAH! That's gonna be a huge compat issue for other plugins, that's a no go as far as I see.